rtmp-client/src/main/cpp/librtmp-jni.c

```diff
@@ -37,17 +33,19 @@ Java_net_butterflytv_rtmp_1client_RtmpClient_nativeOpen(JNIEnv* env, jobject thi
     RTMP *rtmp = (RTMP *) rtmpPointer;
    // rtmp = RTMP_Alloc();
     if (rtmp == NULL) {
-        return -1;
+        throwIllegalStateException(env, "RTMP open called without allocating rtmp object");
+        return RTMP_ILLEGAL_STATE;
```

Comment:
It doesn't really matter what you return here, does it? Maybe call it `RTMP_IGNORED` to make it clearer that the return value is going to be ignored. Ditto other cases where the exception gets thrown.

Reply:
Done.